utils.go

```diff
@@ -798,3 +807,163 @@ func appendLengthEncodedInteger(b []byte, n uint64) []byte {
 	return append(b, 0xfe, byte(n), byte(n>>8), byte(n>>16), byte(n>>24),
 		byte(n>>32), byte(n>>40), byte(n>>48), byte(n>>56))
 }
+
+// escapeBytesBackslash escapes []byte with backslashes (\)
+// This escapes the contents of a string (provided as []byte) by adding backslashes before special
+// characters, and turning others into specific escape sequences, such as
+// turning newlines into \n and null bytes into \0.
+// https://github.com/mysql/mysql-server/blob/mysql-5.7.5/mysys/charset.c#L823-L932
+func escapeBytesBackslash(buf, v []byte) []byte {
+	pos := len(buf)
+	end := pos + len(v)*2
+	if cap(buf) < end {
+		n := make([]byte, pos+end)
+		copy(n, buf)
+		buf = n
+	}
+	buf = buf[0:end]
```

Comment:
The exact same code from L819-L824 is used 4 times. Move it to a helper function. And rename `n` to `newBuf`. `n` is more or less reserved for numbers :wink: